homeassistant/components/persistent_notification/trigger.py

```diff
@@ -0,0 +1,77 @@
+"""Offer persistent_notifications triggered automation rules."""
+from __future__ import annotations
+
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_PLATFORM
+from homeassistant.core import CALLBACK_TYPE, HassJob, HomeAssistant, callback
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.trigger import TriggerActionType, TriggerData, TriggerInfo
+from homeassistant.helpers.typing import ConfigType
+
+from . import Notification, UpdateType, async_register_callback
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ("persistent_notification",)
+
+CONF_NOTIFICATION_ID = "notification_id"
+CONF_UPDATE_TYPE = "update_type"
+
+TRIGGER_SCHEMA = cv.TRIGGER_BASE_SCHEMA.extend(
+    {
+        vol.Required(CONF_PLATFORM): "persistent_notification",
+        vol.Optional(CONF_NOTIFICATION_ID): str,
+        vol.Optional(CONF_UPDATE_TYPE): vol.Coerce(UpdateType),
+    }
+)
+
+
+async def async_attach_trigger(
+    hass: HomeAssistant,
+    config: ConfigType,
+    action: TriggerActionType,
+    trigger_info: TriggerInfo,
+) -> CALLBACK_TYPE:
+    """Listen for state changes based on configuration."""
+    trigger_data: TriggerData = trigger_info["trigger_data"]
+    job = HassJob(action)
+
+    persistent_notification_id = config.get(CONF_NOTIFICATION_ID)
+    persistent_notification_update_type = config.get(CONF_UPDATE_TYPE)
+
+    @callback
+    def persistent_notification_listener(
+        update_type: UpdateType, notifications: dict[str, Notification]
+    ) -> None:
+        """Listen for persistent_notification updates."""
+
+        notification = list(notifications.values())[0]
```

Comment:
What if the one they want isn't the first one?

Reply:
Based on how the `async_dispatcher_send` is called, the notification is always added in a dictionary under the key of its notification_id.
I believe this dictionary will always consist of only one item.

Reply:
https://github.com/home-assistant/core/blob/5834d700375759b760066cdd481575b42a62bc0d/homeassistant/components/persistent_notification/__init__.py#L215

When the websocket api connects it sends the current ones. 

Does the trigger need to know about the current ones when its connected?

Reply:
Did not take that usecase into account.
After some changes It's now going to be just the one notification if only one exists in the dictionary, or a list of notifications if multiple exist.
Also allowed multiple `update_type`s to make it easier to only listen to added / removed callbacks.